kafka-client/src/main/java/dev/responsive/kafka/internal/stores/SessionOperationsImpl.java

```diff
@@ -0,0 +1,285 @@
+
+    switch (params.schemaType()) {
+      case SESSION:
+        return client.sessionTable(params.name().tableName(), partitioner);
+      default:
+        throw new IllegalArgumentException(params.schemaType().name());
+    }
+  }
+
+  private static RemoteSessionTable<?> createRemoteSessionTable(
+      final ResponsiveSessionParams params,
+      final SessionClients sessionClients,
+      final SessionSegmentPartitioner partitioner
+  ) throws InterruptedException, TimeoutException {
+    switch (sessionClients.storageBackend()) {
+      case CASSANDRA:
+        return createCassandra(params, sessionClients, partitioner);
+      case MONGO_DB:
+        return createMongo(params, sessionClients, partitioner);
+      default:
+        throw new IllegalStateException("Unexpected value: " + sessionClients.storageBackend());
+    }
+  }
+
+  @SuppressWarnings("rawtypes")
+  public SessionOperationsImpl(
+      final InternalProcessorContext context,
+      final ResponsiveSessionParams params,
+      final RemoteSessionTable table,
+      final CommitBuffer<SessionKey, ?> buffer,
+      final TopicPartition changelog,
+      final ResponsiveStoreRegistry storeRegistry,
+      final ResponsiveStoreRegistration registration,
+      final ResponsiveRestoreListener restoreListener,
+      final long initialStreamTime
+  ) {
+    this.context = context;
+    this.params = params;
+    this.table = table;
+    this.buffer = buffer;
+    this.changelog = changelog;
+    this.storeRegistry = storeRegistry;
+    this.registration = registration;
+    this.restoreListener = restoreListener;
+    this.initialStreamTime = initialStreamTime;
+  }
+
+  @Override
+  public long initialStreamTime() {
+    return initialStreamTime;
+  }
+
+  @Override
+  public void put(
+      SessionKey key,
+      final byte[] value
+  ) {
+    this.buffer.put(key, value, this.context.timestamp());
+  }
+
+  @Override
+  public void delete(final SessionKey key) {
+    this.buffer.tombstone(key, this.context.timestamp());
+  }
+
+  @Override
+  public byte[] fetch(final SessionKey key) {
+    final Result<SessionKey> localResult = this.buffer.get(key);
+    if (localResult != null) {
+      return localResult.isTombstone ? null : localResult.value;
+    }
+
+    final var remoteResult = this.table.fetch(
+        this.changelog.partition(),
+        key.key,
+        key.sessionStartMs,
+        key.sessionEndMs
+    );
+    return remoteResult;
+  }
+
+  @Override
+  public KeyValueIterator<Windowed<Bytes>, byte[]> fetchAll(
+      final Bytes key,
+      final long earliestSessionEnd,
+      final long latestSessionEnd
+  ) {
+    final SessionKey from = new SessionKey(key, 0, earliestSessionEnd);
+    final SessionKey to = new SessionKey(key, 0, latestSessionEnd);
+
+    final var localResults = this.buffer.all(kv -> {
+      return kv.key.key.equals(key) && kv.key.sessionEndMs >= earliestSessionEnd
+          && kv.key.sessionEndMs <= latestSessionEnd;
```

Comment:
```suggestion
          && kv.key.sessionStartMs <= latestSessionEnd;
```

I'm wondering if we should have a separate integration test where we _don't_ set the CommitBuffer to flush on every record -- sadly the soak won't catch things like this since we don't do any correctness checking at the moment (of course eventually we hope to use your new store-comparison tool for that). It's still probably worth doing an integration test.

Maybe to save time we could knock out all three store types in the same integration test with less-frequent buffer flushing (rather than having an extra test for this in each store type's integration test class)? This would be followup work of course, just thinking out loud. For now, maybe just try running the test once with the buffer flush config commented out, to make sure the results line up (but merging it with the buffer flush config re-enabled)